use (
  {AssertionError} = dep://assert
  {assert} = dep://chai
  expected = ~
)

suite(__filename, proc()
  test("when fulfilled, wrapper must be returned", async proc()
    const w = await(expected(promise.resolve()))
    assert.strictEqual(w.toBeFulfilled(), w)
  end)

  test("when not fulfilled, assertion error must be raised", async proc()
    const w = await(expected(promise.reject()))
    assert.throws(
      proc() expected(w).toBeFulfilled() end
      AssertionError
      "value should have been fulfilled."
    )
  end)
end)
